### PR TITLE
Fix/call tree id generation

### PR DESCRIPTION
**Pull request by gunther-bachmann**

### CODE REVIEW

File: org.testeditor.tcl.dsl/src/main/java/org/testeditor/tcl/dsl/jvmmodel/CallTreeBuilder.xtend [hunk 1]
```diff
@@ -5,41 +5,64 @@ import javax.inject.Singleton
...
 
-			if (!setups.empty) {
-				children += callTreeNodeNamed(testSetupDisplayName) => [
-					children += setups.flatMap[toCallTreeChildren]
+			treeId = 'IDROOT'
+
+			runningNumber = 0
+			idPrefix = TclJvmModelInferrer.ID_PREFIX_CONFIG_SETUP
+			if (!parentSetup.empty) {
+				children += callTreeNodeNamed(#[testConfigName, testSetupDisplayName].join(' ')) => [
```
Comment (jjelschen) [inline org.testeditor.tcl.dsl/src/main/java/org/testeditor/tcl/dsl/jvmmodel/CallTreeBuilder.xtend]: Isn't this overkill just for concatenating two strings, separated by a single space…? Unless I am missing something, I would suggest a more idiomatic approach, e.g.
```suggestion
				children += callTreeNodeNamed('''«testConfigName» «testSetupDisplayName»''')) => [
```
or simply
```suggestion
				children += callTreeNodeNamed(testConfigName + ' ' + testSetupDisplayName)) => [
```

File: org.testeditor.tcl.dsl/src/main/java/org/testeditor/tcl/dsl/jvmmodel/CallTreeBuilder.xtend [hunk 2]
```diff
@@ -64,13 +87,23 @@ class CallTreeBuilder {
 	def dispatch CallTreeNode toCallTree(AbstractTestStep model) {
 		return model.namedCallTreeNode
 	}
+	
+	long runningNumber = 0
 
 	def dispatch CallTreeNode toCallTree(TestStep model) {
-		return model.namedCallTreeNode => [
+		val result = model.namedCallTreeNode
+		result => [
```
Comment (jjelschen) [inline org.testeditor.tcl.dsl/src/main/java/org/testeditor/tcl/dsl/jvmmodel/CallTreeBuilder.xtend:95]: This should also work, right?
```suggestion
		return result => [
```

File: org.testeditor.tcl.dsl/src/main/java/org/testeditor/tcl/dsl/jvmmodel/CallTreeBuilder.xtend [hunk 3]
```diff
@@ -64,13 +87,23 @@ class CallTreeBuilder {
...
+				val previousPrefix = idPrefix
+				val previousRunningNumber = runningNumber
+				idPrefix = result.treeId
+				runningNumber = 0
 				children += model.findMacro.toCallTree
+				idPrefix = previousPrefix
+				runningNumber = previousRunningNumber
 			}
 		]
+		return result
```
Comment (jjelschen) [inline org.testeditor.tcl.dsl/src/main/java/org/testeditor/tcl/dsl/jvmmodel/CallTreeBuilder.xtend:106]:
```suggestion
```

**Review by jjelschen — CHANGES_REQUESTED** [review]

File: org.testeditor.tcl.dsl/src/main/java/org/testeditor/tcl/dsl/jvmmodel/CallTreeBuilder.xtend (hunk 2 above)
Comment (gunther-bachmann) [reply org.testeditor.tcl.dsl/src/main/java/org/testeditor/tcl/dsl/jvmmodel/CallTreeBuilder.xtend:95]: It should. I wanted to make the order of execution explicit (namedCallTreeNode generates a new id and this is explicitly done as a separate step).

**Review by jjelschen — APPROVED** [review]

**Pull request merged**